zerver/lib/upload.py

```diff
@@ -600,24 +599,16 @@ def get_emoji_url(self, emoji_file_name: str, realm_id: int) -> str:
         bucket = settings.S3_AVATAR_BUCKET
         emoji_path = RealmEmoji.PATH_ID_TEMPLATE.format(realm_id=realm_id,
                                                         emoji_file_name=emoji_file_name)
-        return "https://%s.%s/%s" % (bucket, self.connection.DefaultHost, emoji_path)
+        return "https://%s.%s/%s" % (bucket, self.session.region_name, emoji_path)
 
     def upload_export_tarball(self, realm: Optional[Realm], tarball_path: str) -> str:
-        def percent_callback(complete: Any, total: Any) -> None:
-            sys.stdout.write('.')
-            sys.stdout.flush()
-
-        conn = S3Connection(settings.S3_KEY, settings.S3_SECRET_KEY)
-        # We use the avatar bucket, because it's world-readable.
-        bucket = get_bucket(conn, settings.S3_AVATAR_BUCKET)
-        key = Key(bucket)
-        key.key = os.path.join("exports", generate_random_token(32), os.path.basename(tarball_path))
-        key.set_contents_from_filename(tarball_path, cb=percent_callback, num_cb=40)
-
-        public_url = 'https://{bucket}.{host}/{key}'.format(
-            host=conn.server_name(),
-            bucket=bucket.name,
-            key=key.key)
+        key = os.path.join("exports", generate_random_token(32), os.path.basename(tarball_path))
+        s3_object = get_s3_object(settings.S3_AVATAR_BUCKET, key)
+        s3_object.upload_file(tarball_path)
```

Comment:
It looks like that silent failure was [possibly resolved already](https://github.com/boto/boto3/issues/1067#issuecomment-594199212).